lib/pecl

```diff
@@ -0,0 +1,36 @@
+function install_pecl_extension() {
+  local extension_name="${1}"
+  local version="${2}"
+  local cache_dir="${3}"
+
+  local ext_dir=/app/vendor/php/lib/php/extensions/no-debug-non-zts-$(php_api_version)
+
+  local cache_extension_file="${cache_dir}/${extension_name}-${version}.so"
+  if [ -f "${cache_extension_file}" ]; then
+    echo "Installing PECL extension ${extension_name} version ${version} from the cache"
+    cp "${cache_extension_file}" "${ext_dir}/${extension_name}.so"
+    echo "extension=${extension_name}.so" > "/app/vendor/php/etc/conf.d/${extension_name}.ini"
+    return
+  fi
+
+  local build_dir=$(pwd)
+  local temp_dir=$(mktmpdir "pecl-extension")
+
+  echo "Installing PECL extension ${extension_name} version ${version}"
+
+  pushd "${temp_dir}" > /dev/null
+
+  curl --silent --location "https://pecl.php.net/get/${extension_name}-${version}.tgz" | tar xz
+
+  pushd ${extension_name}-${version} > /dev/null
+  /app/vendor/php/bin/phpize > /dev/null
+  ./configure --with-php-config=/app/vendor/php/bin/php-config --with-${extension_name}=${build_dir}/.apt/usr > /dev/null
```

Comment:
The problem is that the option `--with-${extension_name}` is not always useful, or may be named differently. How could we make this configurable? Via an environment variable?

Reply:
I think there's more: some extensions use the `extension=whatever.so` directive while some others use the `zend_extension=whatever.so`.

For now I planned to use a (rather uglyish) `case` statement to handle all this. But I wanted to discuss it also.

That was before the decision to first have infosec have a look at it.

Reply:
There is no way to know if an extension need the zen_extension or the extension directive from PECL (i.e. xdebug needs a zen_extension directive but nothing is mentionned on the pecl repository). I think defining env variables to configure the ` --with-${extension_name}`  and the  `extension` / `zen_extension`  directives could be a solution to this